Files with "!" in the name are not added to content.json

### Step 1: Please describe your environment

  * ZeroNet version: 0.7.0 rev4172
  * Operating system: W7x64
  * Web browser: N/A
  * Tor status: N/A
  * Opened port: N/A
  * Special configuration: No

### Step 2: Describe the problem:

Files that contain `!` (exclamation mark) in the file name are not added to `content.json`.

Note: This seems to apply to a few other characters too, including but not limited to `'` (apostrophe) and `&` (ampersand)

#### Steps to reproduce:

  1. Create blank site
  2. Create a few files with and without an exclamation mark in the name
  3. Set site name and description, then sign and publish the site

#### Observed Results:

When setting the site name and description, the directory is scanned for files, and they are added to the `content.json` index. Files containing an exclamation mark are not added to the file and thus are not shared with others.

The debug log contains lines like these (some information has been redacted by me):

    [2019-08-11 ███] ERROR    Site:███ - [ERROR] Invalid filename: Test !.txt
    [2019-08-11 ███] INFO     Site:███ - [SKIPPED] Test !.txt

#### Expected Results:

No seemingly arbitrarily chosen file name rules in place. Any file name made up of a valid file name character sequence should be added to the `content.json` file.

### Related problems

Files/directories that start with `.` or end in `-old` or `-new` are hardcoded to be excluded too.

This puts a huge dampener on sharing git repositories or creating the `.well-known` folder. (See [RFC 5785](https://tools.ietf.org/html/rfc5785))

### Notes

The offending location for the character selection seems to be https://github.com/HelloZeroNet/ZeroNet/blob/5a746769d01db58d54ff121e784fbf87374a89f3/src/Content/ContentManager.py#L603

--- a/src/Content/ContentManager.py
+++ b/src/Content/ContentManager.py
@@ -600,10 +600,10 @@ class ContentManager(object):
         elif len(relative_path) > 255:
             return False
         else:
-            return re.match(r"^[^\x00-\x1F\x22\x2A\x3A\x3C\x3E\x3F\x5C\x7C]+$", relative_path)
+            return re.match(r"^[^\x00-\x1F\"*:<>?\\|]+$", relative_path)
 
     def sanitizePath(self, inner_path):
-        return re.sub("[\x00-\x1F\x22\x2A\x3A\x3C\x3E\x3F\x5C\x7C]", "", inner_path)
+        return re.sub("[\x00-\x1F\"*:<>?\\|]", "", inner_path)
 
     # Hash files in directory
     def hashFiles(self, dir_inner_path, ignore_pattern=None, optional_pattern=None):

--- a/src/Content/ContentManager.py
+++ b/src/Content/ContentManager.py
@@ -600,10 +600,10 @@ class ContentManager(object):
         elif len(relative_path) > 255:
             return False
         else:
-            return re.match(r"^[a-z\[\]\(\) A-Z0-9~_@=\.\+-/]+$", relative_path)
+            return re.match(r"^[^\x00-\x1F\x22\x2A\x3A\x3C\x3E\x3F\x5C\x7C]+$", relative_path)
 
     def sanitizePath(self, inner_path):
-        return re.sub("[^a-z\[\]\(\) A-Z0-9_@=\.\+-/]", "", inner_path)
+        return re.sub("[\x00-\x1F\x22\x2A\x3A\x3C\x3E\x3F\x5C\x7C]", "", inner_path)
 
     # Hash files in directory
     def hashFiles(self, dir_inner_path, ignore_pattern=None, optional_pattern=None):

--- a/src/Test/TestContent.py
+++ b/src/Test/TestContent.py
@@ -246,3 +246,8 @@ class TestContent:
             with site.storage.open("data/users/1C5sgvWaSgfaTpV5kjBCnCiKtENNMYo69q/content.json") as data:
                 site.content_manager.verifyFile("data/users/1C5sgvWaSgfaTpV5kjBCnCiKtENNMYo69q/content.json", data, ignore_same=False)
         assert "Potentially unsafe" in str(err.value)
+
+
+    @pytest.mark.parametrize("filename", ["test.txt", "test/!@#$%^&().txt", "ÜøßÂŒƂÆÇ.txt"])
+    def testPathValidation(self, site, filename):
+        assert site.content_manager.isValidRelativePath(filename)

--- a/src/Test/TestContent.py
+++ b/src/Test/TestContent.py
@@ -248,6 +248,10 @@ class TestContent:
         assert "Potentially unsafe" in str(err.value)
 
 
-    @pytest.mark.parametrize("filename", ["test.txt", "test/!@#$%^&().txt", "ÜøßÂŒƂÆÇ.txt"])
-    def testPathValidation(self, site, filename):
-        assert site.content_manager.isValidRelativePath(filename)
+    def testPathValidation(self, site):
+        assert site.content_manager.isValidRelativePath("test.txt")
+        assert site.content_manager.isValidRelativePath("test/!@#$%^&().txt")
+        assert site.content_manager.isValidRelativePath("ÜøßÂŒƂÆÇ.txt")
+        assert site.content_manager.isValidRelativePath("тест.текст")
+        assert site.content_manager.isValidRelativePath("𝐮𝐧𝐢𝐜𝐨𝐝𝐞𝑖𝑠𝒂𝒘𝒆𝒔𝒐𝒎𝒆")
+

